### ISSUE TITLE
Spock test does not fail when it should when dealing with big collections

### ISSUE THREAD

**Issue by rizmailov** [spockframework/spock#699]
# Issue description
Spock test does not fail when it should when asserting equality of two collections with size about 2 millions elements.

# How to reproduce
Run this test from Idea
```
void 'should fail'() {
        given:
        def colA = new ArrayList(1)
        when:
        def colB = new ArrayList(2000000)
        for (int i = 0; i < 2000000; i++) {
            colB.add(null)
        }
        then:
        assert colA == colB
    }
```
# Additional Environment information
> Version of your build tool(if used), Java, Groovy, IDE, OS etc
Maven 3.3.9
Java 8 ver 1.8.0_112
Spock 1.0-groovy-2.4
Intellij Idea 2016.3.4
Windows 10

### Apache Maven
    <dependency>
                <groupId>org.spockframework</groupId>
                <artifactId>spock-core</artifactId>
                <version>1.0-groovy-2.4</version>
                <scope>test</scope>
            </dependency>
            <dependency>
                <groupId>org.spockframework</groupId>
                <artifactId>spock-spring</artifactId>
                <version>1.0-groovy-2.4</version>
                <scope>test</scope>
            </dependency>

**Comment by siordache** [comment]
It looks like a bug in the Intellij JUnit plugin. I suspect that the testFailure() method of [JUnit4TestListener](https://github.com/JetBrains/intellij-community/blob/master/plugins/junit_rt/src/com/intellij/junit4/JUnit4TestListener.java) cannot handle the large amount of text involved by the failing condition.

The problem does not occur in Eclipse or when running the test from the command line.

**Comment by szpak** [comment]
@siordache Thanks for sharing your finding.

@rizmailov @siordache Maybe it would be good to report an issue in Idea's issue tracker https://youtrack.jetbrains.com/ and paste it here for further reference.

**Comment by rizmailov** [comment]
@szpak https://youtrack.jetbrains.com/issue/IDEA-168942

**Comment by szpak** [comment]
Thanks @rizmailov . I would add in the request that it works outside Idea.

**Comment by leonard84** [comment]
>Looks like it is fixed in 2017.1
